libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -956,32 +965,29 @@ void PbftManager::firstFinish_() {
   LOG(log_dg_) << "PBFT first finishing state in round: " << round << ", period: " << period;
 
   if (cert_voted_block_for_round_.has_value()) {
-    auto last_cert_voted_block = getUnfinalizedBlock_(cert_voted_block_for_round_->first);
-    assert(last_cert_voted_block != nullptr);
+    const auto [cert_voted_block, cert_voted_block_period] = *cert_voted_block_for_round_;
 
-    blk_hash_t vote_value = last_cert_voted_block->getBlockHash();
-    // If cert_voted_block_for_round_ pushed into chain vote for NULL value
-    if (cert_voted_block_for_round_->second < period) {
-      vote_value = NULL_BLOCK_HASH;
-    }
+    // It should never happen that node moved to the next period without cert_voted_block_for_round_ reset
+    assert(cert_voted_block_period == period);
 
-    if (auto vote = generateVoteWithWeight(vote_value, next_vote_type, period, round, step_); vote) {
-      LOG(log_nf_) << "Placed first finish next vote for " << vote_value << ", vote weight " << *vote->getWeight()
+    if (auto vote = generateVoteWithWeight(cert_voted_block, next_vote_type, cert_voted_block_period, round, step_);
```

Comment:
Why are we using this cert_voted_block_period and not just period?

Reply:
it is the same, see `assert(cert_voted_block_period == period);`.